LDAR_Sim/model_code/methods/deployment/orbit_crew.py

```diff
@@ -0,0 +1,171 @@
+            'remaining_mins': 0,
+        }
+
+    def calc_viewable_sites(self, site_pool):
+        """
+        Subset possible sites that could be surveyed with a given satellite location.
+        Generic utility function to check whether a satellite can see a given patch of ground
+        satstate: dictionary containing satellite lon, lat, and altitude
+        lon: longitude of location to assess
+        lat: latitude of location to assess
+        975 m is the average elevation in Alberta 
+        Returns True to denote the satellite can see the location and False otherwise
+        """
+        valid_site = []
+        # ind = 0
+        sat_date = self.sat_date
+        path = self.orbit_path
+        date = self.state['t'].current_date.date()
+        # find daily pathes
+        DP = path[sat_date == date]
+        for s in site_pool:
+            fac_lat = np.float16(s['lat'])
+            fac_lon = np.float16(s['lon'])
+            PT = Point(fac_lon, fac_lat)
+            for dp in DP:
+                if dp.contains(PT):
+                    valid_site.append(s)
+                    break
+        return valid_site
+
+    def assess_weather(self, site):
+        """
+        Function to perform satellite specific checks of the weather for the purposes of visiting
+        site: the site
+        """
+
+        site_lat = np.float16(site['lat'])
+        site_lon = np.float16(site['lon'])
+
+        lat_idx = geo_idx(site_lat, self.state['weather'].latitude)
+        lon_idx = geo_idx(site_lon, self.state['weather'].longitude)
+        ti = self.state['t'].current_timestep
+
+        # check daylight
+        date = self.state['t'].current_date
+        sr, ss = quick_cal_daylight(date, site_lat, site_lon)
+
+        if sr <= self.state['t'].current_date.hour <= ss:
+            sat_daylight = True
+        else:
+            sat_daylight = False
+        # check cloud cover
+
+        CC = self.cloudcover[ti, lat_idx, lon_idx] * 100
+        CC = round(CC)
+        arr = np.zeros(100)
+        arr[:CC] = 1
+        np.random.shuffle(arr)
+
+        if np.random.choice(arr, 1)[0] == 0:
+            sat_cc = True
+        else:
+            sat_cc = False
+
+        # do some checks and return true or false whether this site checks our weather checks
+        return (sat_daylight, sat_cc)
+
+    def get_orbit_predictor(self):
+        # build a satellite orbit object
+        wd = self.parameters['working_directory']
+        TLEs = []
+        with open(wd+self.tlefile) as f:
+            for line in f:
+                TLEs.append(line.rstrip())
+        i = 0
+        for x in TLEs:
+            if x == self.sat:
+                break
+            i += 1
+        TLE_LINES = (TLEs[i+1], TLEs[i+2])
+        self.predictor = get_predictor_from_tle_lines(TLE_LINES)
+        return
+
+    def get_orbit_path(self):
+
+        #### initiate the orbit path ####
```

Comment:
multiple consecutive # is a pep8 error. I've been replacing with # --- Text  --- for heading comments,  but also not very consistently.